engine/table/src/main/java/io/deephaven/engine/util/PythonScopeJpyImpl.java

```diff
@@ -189,7 +189,16 @@ public static Object convert(PyObject pyObject) {
         } else if (pyObject.isCallable()) {
             return wrapCallable(pyObject);
         } else if (pyObject.isConvertible()) {
-            return pyObject.getObjectValue();
+            Object objValue = pyObject.getObjectValue();
+            // workaround a JPY issue with determining the correct Java type a Python 'int' should be
+            // mapped to (int or long) without causing an implicit overflow
+            if (objValue instanceof Integer || objValue.getClass() == int.class) {
```

Comment:
I'm not a fan of `.getClass()` check here, I don't think it adds anything of value.

Happy to approve without.

I think this convert logic could be pushed all down into python (and save us numerous JNI calls), potentially pending https://github.com/deephaven/deephaven-core/issues/1775. Obviously not part of this simple fix, but something we may want to do in the future. Going to add a note to the linked issue.

Reply:
objValue.getClass() being int.class seems impossible; as you would not have a class object.

Reply:
thanks, removed!